clang/include/clang/Analysis/Analyses/ExprMutationAnalyzer.h

```diff
@@ -22,8 +21,15 @@ class FunctionParmMutationAnalyzer;
 /// a given statement.
 class ExprMutationAnalyzer {
 public:
+  friend class FunctionParmMutationAnalyzer;
+  struct Cache {
+    llvm::DenseMap<const FunctionDecl *,
```

Comment:
Note: consider using llvm::SmallDenseMap, to reduce amount of memory allocations.